cognite/client/_api/annotations.py

```diff
@@ -18,25 +18,32 @@ class AnnotationsAPI(APIClient):
     _RESOURCE_PATH = "/annotations"
 
     @overload
-    def create(self, annotations: Annotation) -> Annotation:
+    def create(self, annotations: Annotation | AnnotationWrite) -> Annotation:
         ...
 
     @overload
-    def create(self, annotations: Sequence[Annotation]) -> AnnotationList:
+    def create(self, annotations: Sequence[Annotation | AnnotationWrite]) -> AnnotationList:
         ...
 
-    def create(self, annotations: Annotation | Sequence[Annotation]) -> Annotation | AnnotationList:
+    def create(
+        self, annotations: Annotation | AnnotationWrite | Sequence[Annotation | AnnotationWrite]
+    ) -> Annotation | AnnotationList:
         """`Create annotations <https://developer.cognite.com/api#tag/Annotations/operation/annotationsCreate>`_
 
         Args:
-            annotations (Annotation | Sequence[Annotation]): Annotation(s) to create
+            annotations (Annotation | AnnotationWrite | Sequence[Annotation | AnnotationWrite]): Annotation(s) to create
 
         Returns:
             Annotation | AnnotationList: Created annotation(s)
         """
-        assert_type(annotations, "annotations", [Annotation, Sequence])
+        assert_type(annotations, "annotations", [AnnotationCore, Sequence])
+
         return self._create_multiple(
-            list_cls=AnnotationList, resource_cls=Annotation, resource_path=self._RESOURCE_PATH + "/", items=annotations
+            list_cls=AnnotationList,
+            resource_cls=Annotation,
+            resource_path=self._RESOURCE_PATH + "/",
+            items=annotations,
+            input_resource_cls=AnnotationWrite,
```

Comment:
This should be the inputed type, no? So, make the signature `Sequence[Annotation] | Sequence[AnnotationWrite]` instead and then `input_resource_cls=type(annotations[0])`.
```suggestion
            input_resource_cls=AnnotationWrite,
```

Reply:
(This applies across the board)

Reply:
I don't think so. The `input_resource_cls` used in the case of an exception to reload the dumped item from the failed request. Since we are now always converting to the write object, we would not be able to load a read instance even if that was the input.